src/actions/index.js

```diff
@@ -1,178 +1,237 @@
+        // we should leave it be, have an option display a null track on posts,
+        // and give them a heads up they haven't set a track with a toast notification... brainstorming
+        history.push('/onboarding');
+      } else {
+        history.push('/');
+      }
+    })
+    .catch((error) => console.log(error));
 };
 
 // Sort out this flow: success, fetchUser, fetchUserProfile
 // react-query will help with making sure user object is in state
 // Other one takes in a user ID, but do we have the user's ID available at all times (for example, when onboarding)
-export const fetchUser = () => dispatch => {
-    axios.get(`${BACKEND_URL}/api/user`)
-        .then(response => {
-            dispatch({ type: 'SET_USER', payload: response.data.user });
-            console.log('FETCH USER ACTION', response.data.user);
-        })
-        .catch(error => console.log(error));
-};
-
-export const logOut = history => dispatch => {
-    axios.get(`${BACKEND_URL}/api/auth/logout`)
-        .then(response => {
-            localStorage.removeItem('id');
-            history.push('/welcome');
-        })
-        .catch(error => console.log(error));
+export const fetchUser = () => (dispatch) => {
+  axios
+    .get(`${BACKEND_URL}/api/user`)
+    .then((response) => {
+      dispatch({ type: 'SET_USER', payload: response.data.user });
+      console.log('FETCH USER ACTION', response.data.user);
+    })
+    .catch((error) => console.log(error));
+};
+
+export const logOut = (history) => (dispatch) => {
+  axios
+    .get(`${BACKEND_URL}/api/auth/logout`)
+    .then((response) => {
+      localStorage.removeItem('id');
+      history.push('/welcome');
+    })
+    .catch((error) => console.log(error));
 };
 
 // Onboarding?
-export const setTrack = (track, token) => dispatch => {
-    return axios.put(`${BACKEND_URL}/api/user/track`, { track, token });
+export const setTrack = (track, token) => (dispatch) => {
+  return axios.put(`${BACKEND_URL}/api/user/track`, { track, token });
 };
 
 // Post
-export const postQuestion = (question, answer, track, category, history) => dispatch => {
-    return (
-        axios.post(`${BACKEND_URL}/api/post/create`, {
-            question: question,
-            answer: answer,
-            track: track,
-            category: category
-        })
-    );
-};
-
-export const fetchSearch = search => dispatch => {
-    axios.post(`${BACKEND_URL}/api/post/search`, { search })
-        .then(response => {
-            console.log("responding", response.data);
-            dispatch({ type: 'SET_POSTS', payload: response.data });
-        })
-        .catch(error => console.log(error));
+export const postQuestion = (question, answer, track, category, history) => (
+  dispatch
+) => {
+  return axios.post(`${BACKEND_URL}/api/post/create`, {
+    question: question,
+    answer: answer,
+    track: track,
+    category: category,
```

Comment:
This category is an artifact from when this project was an interview prep platform. We have removed it from the database, so please remove it from here as well.